CAP/gap/PrintingFunctions.gi

```diff
@@ -23,44 +31,71 @@ InstallGlobalFunction( InfoStringOfInstalledOperationsOfCategory,
+    list_of_potential_algorithmic_properties := SortedList( RecNames( CAP_INTERNAL_CONSTRUCTIVE_CATEGORIES_RECORD ) );
     
-    list_of_algorithmic_properties := Filtered( list_of_mathematical_properties, p -> p in list_of_potential_algorithmic_properties and IsEmpty( CheckConstructivenessOfCategory( category, p ) ) );
+    list_of_algorithmic_and_not_yet_algorithmic_properties := Intersection( list_of_mathematical_properties, list_of_potential_algorithmic_properties );
+    
+    list_of_algorithmic_properties := Filtered( list_of_algorithmic_and_not_yet_algorithmic_properties, p -> IsEmpty( CheckConstructivenessOfCategory( category, p ) ) );
     
     list_of_maximal_algorithmic_properties := MaximalPropertiesWithRegardToImplication( list_of_algorithmic_properties );
     
     StableSortBy( list_of_maximal_algorithmic_properties, p -> Length( CAP_INTERNAL_CONSTRUCTIVE_CATEGORIES_RECORD.( p ) ) );
     
-    result := String( Length( ListPrimitivelyInstalledOperationsOfCategory( category ) ) );
-    result := Concatenation( result, " primitive operations were used to derive " );
-    result := Concatenation( result, String( Length( ListInstalledOperationsOfCategory( category ) ) ) );
-    result := Concatenation( result, " operations for this category" );
     if not IsEmpty( list_of_maximal_algorithmic_properties ) then
+        
         result := Concatenation( result, " which algorithmically" );
+        
         for property in list_of_maximal_algorithmic_properties do
+            
+            result := Concatenation( result, "\n* " );
+            result := Concatenation( result, property );
+            
+        od;
+    fi;
+    
+    list_of_not_yet_algorithmic_properties := Difference( list_of_algorithmic_and_not_yet_algorithmic_properties, list_of_algorithmic_properties );
+    
+    list_of_maximal_not_yet_algorithmic_properties := MaximalPropertiesWithRegardToImplication( list_of_not_yet_algorithmic_properties );
+    
+    StableSortBy( list_of_maximal_not_yet_algorithmic_properties, p -> Length( CAP_INTERNAL_CONSTRUCTIVE_CATEGORIES_RECORD.( p ) ) );
+    
+    if not IsEmpty( list_of_maximal_not_yet_algorithmic_properties ) then
+        
+        if IsEmpty( list_of_maximal_algorithmic_properties ) then
+            result := Concatenation( result, " which" );
+        else
+            result := Concatenation( result, "\nand" );
+        fi;
+        
+        result := Concatenation( result, " not yet algorithmically" );
+        
+        for property in list_of_maximal_not_yet_algorithmic_properties do
             result := Concatenation( result, "\n* " );
             result := Concatenation( result, property );
         od;
+        
     fi;
     
-    list_of_non_algorithmic_mathematical_properties := Difference( list_of_mathematical_properties, list_of_algorithmic_properties );
+    list_of_non_algorithmic_mathematical_properties :=
```

Comment:
Yes, of course, those should be consistent. If you think the name gets too long, we can keep the current names and just add a suitable comment.

Reply:
No, this is OK.

Reply:
> and/or add a small comment here which explains why we consider all implications here.

Just to be sure. Do we need a comment after the renaming?

Reply:
> > and/or add a small comment here which explains why we consider all implications here.
> 
> Just to be sure: Do we need a comment after the renaming?

Not strictly need, but explaining the rationale behind this change would certainly help in the future -> your decision.

Reply:
The comment should not explain *what* is done but *why* it is done.

Reply:
```
## Finally, capture the remaining mathematical properties that influence the derivation mechanism:
```

Reply:
This could as well have been a comment in the old code.
So it's correct and certainly nice to have, but the whole point of this PR was turning
```
list_of_non_algorithmic_mathematical_properties := Difference( list_of_mathematical_properties, list_of_algorithmic_properties );
```
into
```
list_of_mathematical_properties_not_implied_by_algorithmic_or_not_yet_algorithmic_properties :=
      Difference( list_of_mathematical_properties,
              Concatenation( List( list_of_algorithmic_and_not_yet_algorithmic_properties, p -> ListImpliedFilters( ValueGlobal( p ) ) ) ) );
```
The rationale behind this change is what needs an explanation.

Reply:
Ah, I now understand what you mean.

Reply:
The whole idea of the change was to filter out from the list "mathematical properties influencing the derivation mechanism" more than the `list_of_algorithmic_properties`, namely everything implied by any algorithmic or non-algorithmic property. I think this is now reflected in the long name below.

Reply:
As discussed yesterday: `InfoStringOfInstalledOperationsOfCategory` is expected to suppress the listing of properties implied by others (i.e., only list the "maximal" properties). For example, it should never list `IsAbCategory` if `IsAdditiveCategory` is listed. This PR improves `InfoStringOfInstalledOperationsOfCategory` along this line.